src/engine/IndexScan.h

```diff
@@ -168,6 +168,15 @@ class IndexScan final : public Operation {
 
   Permutation::Enum permutation() const { return permutation_; }
 
+  // If this IndexScan is configured to return only a subset of its columns,
+  // return that subset as column indices; otherwise return std::nullopt.
+  std::optional<std::vector<ColumnIndex>> getColumnSubsetIfAny() const {
+    if (varsToKeep_.has_value()) {
+      return getSubsetForStrippedColumns();
+    }
+    return std::nullopt;
+  }
+
```

Comment:
see above, some of these functions are not needed anymore when you incorporate my suggestions.